lib/Parse/ParseDecl.cpp

```diff
@@ -8571,6 +8571,11 @@ parseDeclDeinit(ParseDeclOptions Flags, DeclAttributes &Attributes) {
       diagnose(LParenLoc, diag::opening_paren);
     }
   };
+    
+  SourceLoc asyncLoc, throwsLoc;
+  parseEffectsSpecifiers(SourceLoc(),
+                         asyncLoc, /*reasync=*/nullptr,
+                         throwsLoc, /*rethrows=*/nullptr);
```

Comment:
You will also need to update the new parser to accept an `async` specifier after `deist`.

Reply:
Draft - https://github.com/apple/swift-syntax/pull/1656